core/src/main/scala/org/apache/spark/eventhubs/ConnectionStringBuilder.scala

```diff
@@ -240,6 +240,10 @@ class ConnectionStringBuilder private () {
     this
   }
 
+  def getNamespace: String = {
+    this.getEndpoint.getHost.dropRight(".servicebus.windows.net".length)
```

Comment:
".servicebus.windows.net" is not the only available domain. The assumption that all namespaces are in this domain is incorrect. Also, no value should be hardcoded in a method.